Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ActionManager/Action/ActionManager.h>
+
+namespace AzToolsFramework
+{
+    ActionManager::ActionManager()
+    {
+        AZ::Interface<ActionManagerInterface>::Register(this);
+    }
+
+    ActionManager::~ActionManager()
+    {
+        AZ::Interface<ActionManagerInterface>::Unregister(this);
+
+        ClearActionContextMap();
+    }
+
+    ActionManagerOperationResult ActionManager::RegisterActionContext(
+        QWidget* widget,
+        AZStd::string_view identifier,
+        AZStd::string_view name,
+        AZStd::string_view parentIdentifier)
+    {
+        if (!widget)
+        {
+            return AZ::Failure(
+                AZStd::string::format("Action Manager - Could not register action context \"%s\" to a null widget.", identifier.data())
```

Comment:
You can't do this sorry 😮 `.data()` might not be null terminated, so this could go horribly wrong. There is a way to do this for format strings - `"%.*s", stringLength, pointerToString`, but be super careful just calling `.data()` without a size (I'll send you an article to it offline 😝)